@coderabbitai Latest addon checker logs:
```
Run xbmc/action-kodi-addon-checker@v1.2
  with:
    kodi-version: matrix
    is-pr: false
    addon-id: script.cabertoss
    rewrite-for-matrix: false
/usr/bin/docker run --name fa03d61e5081e1157410eae6740d6df2a873f_410cc8 --label 7fa03d --workdir /github/workspace --rm -e "INPUT_KODI-VERSION" -e "INPUT_IS-PR" -e "INPUT_ADDON-ID" -e "INPUT_REWRITE-FOR-MATRIX" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e "ACTIONS_RESULTS_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/script.cabertoss/script.cabertoss":"/github/workspace" 7fa03d:61e5081e1157410eae6740d6df2a873f  "matrix" "script.cabertoss" "false" "false"
Traceback (most recent call last):
  File "/usr/local/bin/kodi-addon-checker", line 5, in <module>
    from kodi_addon_checker.__main__ import main
  File "/usr/local/lib/python3.7/site-packages/kodi_addon_checker/__main__.py", line 14, in <module>
    from kodi_addon_checker import __version__, check_addon, ValidKodiVersions
  File "/usr/local/lib/python3.7/site-packages/kodi_addon_checker/check_addon.py", line 101
    if (extension := parsed_xml.find('extension')) and extension.get('point') == 'kodi.resource.language':
                  ^
SyntaxError: invalid syntax
```

--- a/resources/language/resource.language.en_gb/strings.po
+++ b/resources/language/resource.language.en_gb/strings.po
@@ -52,3 +52,7 @@ msgstr "
 msgctxt "#32030"
 msgid "Working..."
 msgstr "
+
+msgctxt "#32031"
+msgid "Error making directory to copy log files to"
+msgstr "

--- a/resources/lib/cabertoss.py
+++ b/resources/lib/cabertoss.py
@@ -90,10 +90,19 @@ def gather_log_files():
 
 def copy_log_files(log_files: List) -> bool:
     """
-    Actually copy the log files to the path in the addon settings
+    Copy the provided Kodi log files into a timestamped destination folder under the configured addon destination.
 
-    @param log_files: List list of log files to copy
-    @return bool: indicating success or failure
+    Detailed behavior:
+    - Expects log_files as a list of 2-element entries [type, path], where `type` is e.g. 'log', 'oldlog', or 'crashlog' and `path` is the source filesystem path.
+    - Creates a destination directory at Store.destination_path named "<hostname>_Kodi_Logs_<YYYY-MM-DD_HH-MM-SS>".
+    - For entries with type 'log' or 'oldlog', reads the source, sanitizes the content with clean_log() (because these paths may be URLs with embedded user/password details), and writes the sanitized content to a file with the same basename in the destination folder.
+    - For other types (e.g., crash logs), copies the source file to the destination folder unchanged.
+
+    Parameters:
+        log_files (List): list of log descriptors [type, path] to copy.
+
+    Returns:
+        bool: True if files were successfully copied, False otherwise.
     """
     if not log_files:
         Logger.error(LANGUAGE(32025))
@@ -128,6 +137,17 @@ def copy_log_files(log_files: List) -> bool:
 
 # This is 'main'...
 def run():
+    """
+    Run the log collection and copying flow: initialize this addon's logging, load configuration, gather Kodi log files, copy them to the configured destination, notify the user, and stop this addon's logging.
+
+    This function performs the module's main orchestration. It:
+    - Starts the logger for this addon's internal logging (not Kodi's general logging system) and loads addon configuration from settings.
+    - If no destination path is configured, shows an error notification and skips copying.
+    - Otherwise, notifies the user, gathers available log files, attempts to copy them to the configured destination, and notifies success (including number of files copied) or failure.
+    - Stops this addon's internal logging before returning.
+
+    Side effects: starts/stops this addon's internal logging, reads configuration, performs filesystem operations (reading, sanitizing, and copying log files), and shows user notifications. Returns None.
+    """
     Logger.start()
     Store.load_config_from_settings()
 

--- a/resources/lib/cabertoss.py
+++ b/resources/lib/cabertoss.py
@@ -13,7 +13,7 @@ from bossanova808.constants import LANGUAGE
 from bossanova808.logger import Logger
 from bossanova808.notify import Notify
 from resources.lib.store import Store
-from resources.lib.clean import *
+from resources.lib.clean import clean_log
 
 
 def gather_log_files():
@@ -24,9 +24,9 @@ def gather_log_files():
     """
 
     # Basic log files
-    log_files = [['log', os.path.join(LOG_PATH, 'kodi.log')]]
+    log_files = [('log', os.path.join(LOG_PATH, 'kodi.log'))]
     if os.path.exists(os.path.join(LOG_PATH, 'kodi.old.log')):
-        log_files.append(['oldlog', os.path.join(LOG_PATH, 'kodi.old.log')])
+        log_files.append(('oldlog', os.path.join(LOG_PATH, 'kodi.old.log')))
 
     # Can we find a crashlog?
     # @TODO - add Android support if possible..?
@@ -80,7 +80,7 @@ def gather_log_files():
         if lastcrash:
             # Logger.info(f"lastcrash {lastcrash}")
             for crashfile in lastcrash:
-                log_files.append(['crashlog', crashfile])
+                log_files.append(('crashlog', crashfile))
 
     Logger.info("Found these log files to copy (type, basename):")
     Logger.info([[t, os.path.basename(p)] for t, p in log_files])
@@ -114,7 +114,10 @@ def copy_log_files(log_files: List[Tuple[str, str]]) -> bool:
 
     try:
         Logger.info(f'Making destination folder: {now_destination_path}')
-        xbmcvfs.mkdir(now_destination_path)
+        if not xbmcvfs.mkdir(now_destination_path):
+            Logger.error(f'Failed to create destination folder: {now_destination_path}')
+            Notify.error(LANGUAGE(32031))
+            return False
         for file in log_files:
             if file[0] in ['log', 'oldlog']:
                 Logger.info(f'Copying sanitised {file[0]} {file[1]}')
@@ -122,8 +125,11 @@ def copy_log_files(log_files: List[Tuple[str, str]]) -> bool:
                     content = current.read()
                     sanitised = clean_log(content)
                 dest_path = os.path.join(xbmcvfs.translatePath(now_destination_path), os.path.basename(file[1]))
-                with xbmcvfs.File(dest_path, 'w') as output:
-                    output.write(sanitised.encode('utf-8'))
+                f = xbmcvfs.File(dest_path, 'w')
+                try:
+                    f.write(sanitised.encode('utf-8'))
+                finally:
+                    f.close()
             else:
                 Logger.info(f'Copying {file[0]} {file[1]}')
                 if not xbmcvfs.copy(file[1], os.path.join(now_destination_path, os.path.basename(file[1]))):
@@ -162,7 +168,7 @@ def run():
             if result:
                 Notify.info(LANGUAGE(32028) + f": {len(log_file_list)}")
             else:
-                Notify.info(LANGUAGE(32029))
+                Notify.error(LANGUAGE(32029))
     # and, we're done...
     finally:
         Logger.stop()

--- a/resources/lib/cabertoss.py
+++ b/resources/lib/cabertoss.py
@@ -1,7 +1,5 @@
 # -*- coding: utf-8 -*-
 import os
-from datetime import datetime
-from time import sleep
 from datetime import datetime, timedelta
 import socket
 from typing import List, Tuple
@@ -14,6 +12,12 @@ from bossanova808.logger import Logger
 from bossanova808.notify import Notify
 from resources.lib.store import Store
 from resources.lib.clean import clean_log
+
+
+def _vfs_join(base: str, name: str) -> str:
+    if base.startswith(('special://', 'smb://', 'nfs://', 'ftp://', 'http://', 'https://')):
+        return base.rstrip('/') + '/' + name
+    return os.path.join(base, name)
 
 
 def gather_log_files():
@@ -51,7 +55,7 @@ def gather_log_files():
         filematch = 'kodi_'
     elif xbmc.getCondVisibility('system.platform.android'):
         Logger.info("System is Android")
-        Logger.info(LANGUAGE(32023))
+        Logger.info(LANGUAGE(32024))
 
     # If *ELEC, we can be more specific
     if xbmc.getCondVisibility('System.HasAddon(service.coreelec.settings)') or xbmc.getCondVisibility('System.HasAddon(service.libreelec.settings)'):
@@ -70,7 +74,7 @@ def gather_log_files():
                     if three_days_ago < datetime.fromtimestamp(os.path.getmtime(item_with_path)):
                         items.append(os.path.join(crashlog_path, item))
 
-        items.sort(key=lambda f: os.path.getmtime(f))
+        items.sort(key=lambda f:os.path.getmtime(f))
         # Windows crashlogs are a dmp and stacktrace combo...
         if xbmc.getCondVisibility('system.platform.windows'):
             lastcrash = items[-2:]
@@ -114,7 +118,7 @@ def copy_log_files(log_files: List[Tuple[str, str]]) -> bool:
 
     try:
         Logger.info(f'Making destination folder: {now_destination_path}')
-        if not xbmcvfs.mkdir(now_destination_path):
+        if not xbmcvfs.mkdirs(now_destination_path):
             Logger.error(f'Failed to create destination folder: {now_destination_path}')
             Notify.error(LANGUAGE(32031))
             return False
@@ -124,7 +128,7 @@ def copy_log_files(log_files: List[Tuple[str, str]]) -> bool:
                 with open(xbmcvfs.translatePath(file[1]), 'r', encoding='utf-8', errors='replace') as current:
                     content = current.read()
                     sanitised = clean_log(content)
-                dest_path = os.path.join(xbmcvfs.translatePath(now_destination_path), os.path.basename(file[1]))
+                dest_path = _vfs_join(now_destination_path, os.path.basename(file[1]))
                 f = xbmcvfs.File(dest_path, 'w')
                 try:
                     f.write(sanitised.encode('utf-8'))
@@ -132,7 +136,7 @@ def copy_log_files(log_files: List[Tuple[str, str]]) -> bool:
                     f.close()
             else:
                 Logger.info(f'Copying {file[0]} {file[1]}')
-                if not xbmcvfs.copy(file[1], os.path.join(now_destination_path, os.path.basename(file[1]))):
+                if not xbmcvfs.copy(file[1], _vfs_join(now_destination_path, os.path.basename(file[1]))):
                     return False
         return True
 

--- a/resources/lib/clean.py
+++ b/resources/lib/clean.py
@@ -11,6 +11,8 @@ def clean_log(content: str) -> str:
     replaces = (
         # Replace only the credentials part between '//' and '@'
         (r'(?<=//)([^/@:\s]+):([^/@\s]+)@', r'USER:PASSWORD@'),
+        # Also scrub username only (no password)
+        (r'(?<=//)([^/@:\s]+)@', r'USER@'),
         # Replace XML username/password; keep it local to the tag
         (r'(?is)<user>[^<]*</user>', r'<user>USER</user>'),
         (r'(?is)<pass>[^<]*</pass>', r'<pass>PASSWORD</pass>'),

--- a/resources/lib/store.py
+++ b/resources/lib/store.py
@@ -29,6 +29,8 @@ class Store:
         Reads the 'log_path' setting and assigns it to Store.destination_path, then logs the resolved path (sanitized with clean_log because these paths may be URLs with embedded user/password details). This is called at startup and when settings are reloaded; it has no return value.
         """
         Logger.info("Loading configuration from settings")
-        Store.destination_path = ADDON.getSetting('log_path')
-
-        Logger.info(f'Logs will be tossed to: {clean_log(Store.destination_path)}')
+        Store.destination_path = ADDON.get_setting('log_path')
+        if Store.destination_path:
+            Logger.info(f'Logs will be tossed to: {clean_log(Store.destination_path)}')
+        else:
+            Logger.warning(f'No path set to toss logs to.')

--- a/resources/lib/store.py
+++ b/resources/lib/store.py
@@ -1,6 +1,6 @@
-from bossanova808.constants import *
+from bossanova808.constants import ADDON
 from bossanova808.logger import Logger
-from resources.lib.clean import *
+from resources.lib.clean import clean_log
 
 
 class Store:

--- a/resources/lib/store.py
+++ b/resources/lib/store.py
@@ -24,8 +24,9 @@ class Store:
     @staticmethod
     def load_config_from_settings():
         """
-        Load in the addon settings, at start or reload them if they have been changed
-        Log each setting as it is loaded
+        Load the addon's configuration from persistent settings.
+
+        Reads the 'log_path' setting and assigns it to Store.destination_path, then logs the resolved path (sanitized with clean_log because these paths may be URLs with embedded user/password details). This is called at startup and when settings are reloaded; it has no return value.
         """
         Logger.info("Loading configuration from settings")
         Store.destination_path = ADDON.getSetting('log_path')

